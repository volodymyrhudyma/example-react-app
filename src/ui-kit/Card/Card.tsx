import React, { ReactNode } from 'react';
import { Wrapper } from './styled/Card.styled';

interface ICardProps {
  width: string;
  height: string;
  children: ReactNode;
  padding: string;
}

const Card = ({ children, ...rest }: ICardProps) => (
  <Wrapper {...rest}>{children}</Wrapper>
);

Card.defaultProps = {
  width: 'auto',
  height: 'auto',
  padding: '3rem',
};

export default Card;
