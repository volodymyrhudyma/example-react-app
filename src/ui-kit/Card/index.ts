import Component from './Card';

export default Component;
