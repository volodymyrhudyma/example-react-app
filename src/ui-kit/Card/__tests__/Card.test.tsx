import React from 'react';
import { shallow } from 'enzyme';
import Card from '../Card';

describe('Card component', () => {
  it('should render correctly', () => {
    const component = shallow(<Card>Card</Card>);
    expect(component).toMatchSnapshot();
  });
});
