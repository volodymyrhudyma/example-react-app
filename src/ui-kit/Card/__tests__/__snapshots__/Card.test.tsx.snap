// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should render correctly 1`] = `
<styled.div
  height="auto"
  padding="3rem"
  width="auto"
>
  Card
</styled.div>
`;
